posts/napari-cosmx-basics/using-napari-cosmx.qmd

```diff
@@ -0,0 +1,501 @@
+
+fig_path = output_path + "/fig-contours.png"
+with imageio.get_writer(fig_path, dpi=(800, 800)) as writer:
+        screenshot = gem.viewer.screenshot(canvas_only=True)
+        writer.append_data(screenshot)
+
+## White background
+gem.viewer.window.qt_viewer.canvas.background_color_override = 'white'
+
+fig_path = output_path + "/fig-white.png"
+with imageio.get_writer(fig_path, dpi=(800, 800)) as writer:
+        screenshot = gem.viewer.screenshot(canvas_only=True)
+        writer.append_data(screenshot)
+
+
+```
+
+In practice, I use the GUI to adjust the settings (_e.g._, zoom, opacity) and then
+"jot down" the results in my text editor. For example, when I zoom or pan to another location, 
+that location can be found at:
+
+```{r}
+#| eval: false
+gem.viewer.camera.zoom
+gem.viewer.camera.center
+```
+
+Similarly, the contrast limits and gamma values for IF channels can be saved as well. 
+
+```{r}
+#| eval: false
+dna = gem.viewer.layers['DNA']
+dna.contrast_limits = [507.82222222222225, 1260.8000000000002]
+dna.gamma = 1.1682758620689655
+```
+
+Screenshots can be done programmatically with the napari's `screenshot` method
+and there are additional settings you can change (_e.g._, just the canvas, scale)
+that we won't cover here.
+
+```{r}
+#| eval: false
+fig_path = output_path + "/fig-transcripts.png"
+with imageio.get_writer(fig_path, dpi=(800, 800)) as writer:
+        screenshot = gem.viewer.screenshot(canvas_only=False)
+        writer.append_data(screenshot)
+```
+
+There are also methods available in `napari-cosmx` that do not have the GUI
+equivalent. We won't be able to touch on all of these methods in this post but I 
+want to highlight a few. 
+
+## Plot transcripts with an expanded color pallette
+
+The GUI offers a handful of colors to plot transcripts. We can specify which color,
+by name or by hexcode, to plot. For example: 
+
+```{r}
+#| eval: false
+gem.plot_transcripts(gene = "Calb1", color = 'pink', point_size=20)
+```
+
+which is the same as
+
+```{r}
+#| eval: false
+gem.plot_transcripts(gene = "Calb1", color = '#FFC0CB', point_size=20)
+```
+
+## Plotting genes with list comprehensions
+
+We can plot similar genes or targets with the same color. For example, here's
+the negatives in white.
+
+```{r}
+#| eval: false
+#| 
+df = gem.targets
+filtered_df = df[df.target.str.contains("NegPrb")]
+
+pandas_df = filtered_df.to_pandas_df()
+negatives = pandas_df.target.unique().tolist()
+[gem.plot_transcripts(gene = x, color = "white", point_size=10) for x in negatives];
+
+
+```
+
+We can also combine targets with a list of tuples:
+
+```{r}
+#| eval: false
+genes = [('Npy', "orange"), ("Calb1", "cyan")]
+[gem.plot_transcripts(gene = x[0], color = x[1], point_size=10) for x in genes];
+
+```
+
+## Changing transcript transparency
+
+Sometimes transcripts can be stacked on top of each other to the point you cannot
+discern any features. For example (to do). Adjusting the transparents of the layer
```

Comment:
What example?
One can change the layer opacity as a whole in GUI as well, why you say it only change one single point?
Smaller point size is the most common solution to crowding issue and you may want to mention that in addition to change layer opacity.